gateway-service-impl/src/test/java/org/hypertrace/gateway/service/baseline/BaselineServiceImplTest.java

```diff
@@ -92,6 +96,58 @@ public void testBaselineForEntitiesForAggregates() {
         baselineResponse.getBaselineEntityList().get(0).getBaselineAggregateMetricCount() > 0);
   }
 
+  @Test
+  public void testBaselineEntitiesForAggregatesForAvgRateFunction() {
+    BaselineEntitiesRequest baselineEntitiesRequest =
+        BaselineEntitiesRequest.newBuilder()
+            .setEntityType("SERVICE")
+            .setStartTimeMillis(Instant.parse("2020-11-14T17:40:51.902Z").toEpochMilli())
+            .setEndTimeMillis(Instant.parse("2020-11-14T18:40:51.902Z").toEpochMilli())
+            .addEntityIds("entity-1")
+            .addBaselineAggregateRequest(
+                getFunctionExpressionForAvgRate(
+                    FunctionType.AVGRATE, "SERVICE.duration", "duration_ts"))
+            .build();
+
+    // Mock section
+    AttributeMetadata attributeMetadata =
+        AttributeMetadata.newBuilder().setFqn("Service.Latency").setId("Service.StartTime").build();
+    Mockito.when(
+            attributeMetadataProvider.getAttributeMetadata(
+                Mockito.any(RequestContext.class), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(Optional.of(attributeMetadata));
+    Mockito.when(
+            baselineServiceQueryExecutor.executeQuery(
+                Mockito.anyMap(), Mockito.any(QueryRequest.class)))
+        .thenReturn(getResultSet().iterator());
+    when(entityIdColumnsConfigs.getIdKey("SERVICE")).thenReturn(Optional.of("id"));
+    // Attribute Metadata map contains mapping between Attributes and ID to query data. 
+    Map<String, AttributeMetadata> attributeMap = new HashMap<>();
+    attributeMap.put(
+        "SERVICE.duration",
+        AttributeMetadata.newBuilder().setFqn("Service.Latency").setId("Service.Id").build());
+    Mockito.when(
+            attributeMetadataProvider.getAttributesMetadata(
+                Mockito.any(RequestContext.class), Mockito.anyString()))
+        .thenReturn(attributeMap);
+
+    BaselineService baselineService =
+        new BaselineServiceImpl(
+            attributeMetadataProvider,
+            baselineServiceQueryParser,
+            baselineServiceQueryExecutor,
+            entityIdColumnsConfigs);
+    BaselineEntitiesResponse baselineResponse =
+        baselineService.getBaselineForEntities(TENANT_ID, baselineEntitiesRequest, Map.of());
+    Assertions.assertTrue(baselineResponse.getBaselineEntityCount() > 0);
+    Assertions.assertTrue(
+        baselineResponse.getBaselineEntityList().get(0).getBaselineAggregateMetricCount() > 0);
+    BaselineEntity baselineEntity = baselineResponse.getBaselineEntityList().get(0);
+    // verify the baseline for AVG RATE (medianValue/60)
+    Assertions.assertEquals(1.0,
```

Comment:
How is this assertion on double working. I always thought for double we need give a delta too
`assertEquals(double expected, double actual, double delta)`

Reply:
Looks like in Junit5 it is improved and delta is optional. double value converted to long bits and comparison is happening.